apiserver/common/instancetypes.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package common
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/constraints"
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/environs/config"
+	"github.com/juju/juju/environs/instances"
+)
+
+// InstanceTypeBackend provides the methods necessary to obtain provider
+// information.
+type InstanceTypeBackend interface {
+	environs.EnvironConfigGetter
+	Close() error
```

Comment:
Unused, drop me. I don't think it's appropriate to hand off ownership/responsibility to InstanceTypes anyway.